### PR TITLE
Scripts/Items: Goblin Gumbo Kettle

### PR DESCRIPTION
**Pull request by DanVS**
**Changes proposed:**
Implement [Goblin Gumbo Kettle](http://www.wowhead.com/item=33219) buff that causes player to occasionally belch green smoke. I did not find exact timings, so I chose 10 to 45 seconds long trigger period.

**Target branch(es):** 3.3.5

**Tests performed:** Does build and works in-game.

### CODE REVIEW

File: src/server/scripts/Spells/spell_item.cpp [hunk 1]
```diff
@@ -706,6 +706,35 @@ class spell_item_defibrillate : public SpellScriptLoader
...
+{
+    PrepareAuraScript(spell_item_goblin_gumbo);
+
+    void CalcPeriodic(AuraEffect const* /*aurEff*/, bool& isPeriodic, int32& amplitude)
+    {
+        isPeriodic = true;
+        amplitude = urandms(10, 45);
+    }
+
+    void Update(AuraEffect* effect)
```
Comment (illfated) [inline src/server/scripts/Spells/spell_item.cpp]:
```cpp
src/server/scripts/Spells/spell_item.cpp:725:29:
 fatal error: unused parameter 'effect' [-Wunused-parameter]
    void Update(AuraEffect* effect)
                            ^
```
`/*effect*/`

**Comment by illfated** [comment]
Cosmetic, but still:
```sql
DELETE FROM `spell_script_names` WHERE `spell_id`=42760;
INSERT INTO `spell_script_names` VALUES(42760,'spell_item_goblin_gumbo');
```
Suggestion: either add a space after `VALUES` or move the values to the next line.
(I chose not to comment in code, because comments below last line don't get hidden after a change.)

**Comment by DanVS** [comment]
Fixed. Thanks!

**Comment by illfated** [comment]
Sorry, forgot a detail. Regarding
```sql
DELETE FROM `spell_script_names` WHERE `spell_id`=42760;
```
I forgot to mention that TC always deletes by `ScriptName` , not by `spell_id` .

**Comment by illfated** [comment]
Ah, I must have been very tired a week ago. 😫
Otherwise I would have suggested correcting the PR title too:

**Scripts/Items: Goblin Gumbo Kettle**

("Core" is usually code from /src/server/game/ or server runtime related)

**Comment by DanVS** [comment]
Okay, fixed.

File: src/server/scripts/Spells/spell_item.cpp [hunk 2]
```diff
@@ -706,6 +706,35 @@ class spell_item_defibrillate : public SpellScriptLoader
...
+
+// 42760 - Goblin Gumbo
+class spell_item_goblin_gumbo : public AuraScript
+{
+    PrepareAuraScript(spell_item_goblin_gumbo);
+
+    void CalcPeriodic(AuraEffect const* /*aurEff*/, bool& isPeriodic, int32& amplitude)
+    {
+        isPeriodic = true;
+        amplitude = urandms(10, 45);
```
Comment (Shauren) [inline src/server/scripts/Spells/spell_item.cpp]: This is calculated only once on aura apply, was this your intention?

File: src/server/scripts/Spells/spell_item.cpp [hunk 3]
```diff
@@ -706,6 +706,35 @@ class spell_item_defibrillate : public SpellScriptLoader
...
+    }
+
+    void Update(AuraEffect* /*effect*/)
+    {
+        if (Unit* owner = GetUnitOwner())
+            owner->CastSpell(owner, SPELL_GOBLIN_GUMBO_VISUAL, false);
```
[the remark this answers is not in the document]
Comment (Keader) [reply src/server/scripts/Spells/spell_item.cpp]: And missing validate  ```SPELL_GOBLIN_GUMBO_VISUAL```

**Comment by DanVS** [comment]
Fixed. Thanks!

**Comment by mdX7** [comment]
Theres no script needed for this. It seems like its all about the procflag used by 42760(buff caused by item). In the videos I watched about this spell its always a cooldown of 20s, which fits the proc cooldown. 
The procflag 0x1 does not seem to be PROC_FLAG_KILLED, its more likely something like PROC_ON_TICK. Also checked 42556 Headless Horseman Climax - Body Regen (removed on death) in sniffs, the proc in this spell is also always casted every tick. Tickrate seems to be 1s on retail. The other spells having this procflag and SPELL_AURA_PROC_TRIGGER_SPELL seem to be debug spells/removed spells.

**Pull request closed without merging**